packages/grid/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx

```diff
@@ -36,16 +36,15 @@ const GridVirtualScroller = React.forwardRef<
   HTMLDivElement,
   React.HTMLAttributes<HTMLDivElement> & { sx?: SxProps<Theme> }
 >(function GridVirtualScroller(props, ref) {
-  const { className, ...other } = props;
   const rootProps = useGridRootProps();
   const classes = useUtilityClasses(rootProps);
 
   return (
     <VirtualScrollerRoot
       ref={ref}
-      className={clsx(classes.root, className)}
+      {...props}
+      className={clsx(classes.root, props.className)}
       ownerState={rootProps}
-      {...other}
     />
   );
```

Comment:
I believe there are a good number of cases in Material UI where we could apply this optimization. To consider if it's worth doing.

Reply:
Yes, we're doing a lot of allocations. There is also [this issue](https://github.com/mui/material-ui/pull/37461#issuecomment-1592131330) with the spread operator.